projects/addon-editor/components/editor/editor.component.ts

```diff
@@ -133,8 +136,19 @@ export class TuiEditorComponent
         return ``;
     }
 
-    private readonly isSelectionLink = ({startContainer, endContainer}: Range): boolean =>
-        !!startContainer.parentElement?.closest(`a`)?.contains(endContainer);
+    private readonly isSelectionLink = (range: Range): boolean =>
+        this.currentFocusedNodeIsAnchor(range) && this.isSafeRangeForTextNode(range);
+
+    private currentFocusedNodeIsAnchor(range: Range): boolean {
+        return !!range.startContainer.parentElement
```

Comment:
Can this just be `!!range.commonAncestorContainer.closest('a')`?

Reply:
I tried to use this but for some reason the range object returns the previous text element in commonAncestorContainer. Though at the same time, when the cursor is on another line, there should have been a new element.